perfkitbenchmarker/providers/gcp/gcp_dpb_dataflow.py

```diff
@@ -152,4 +186,303 @@ def GetMetadata(self):
+    )
+
+    aggregation = Aggregation(
+        {
+            "alignment_period": {"seconds": 60},  # 1 minute
+            "per_series_aligner": Aggregation.Aligner.ALIGN_MEAN,
+            "cross_series_reducer": Aggregation.Reducer.REDUCE_MEAN,
+            "group_by_fields": ["resource.instance_id"],
+        }
+    )
+
+    results = client.list_time_series(
+      request={
+        "name": project_name,
+        "filter": api_filter,
+        "interval": interval,
+        "view": monitoring_v3.ListTimeSeriesRequest.TimeSeriesView.FULL,
+        "aggregation": aggregation,
+      }
+    )
+
+    # if not results:
+    #   raise Exception('No monitoring data found. Unable to calculate avg CPU utilization')
+    return self._GetAvgValueFromTimeSeries(results)
+
+  def GetMaxOutputThroughput(self, ptransform: str, start_time: datetime, end_time: datetime):
+    """Get max throughput from a particular pTransform during job run interval.
+
+    Args:
+      start_time: datetime specifying the beginning of the time interval.
+      end_time: datetime specifying the end of the time interval.
+
+    Returns:
+      Max value across time interval
+    """
+    client = monitoring_v3.MetricServiceClient()
+    project_name = f"projects/{self.project}"
+    
+    now_seconds = int(time.time())
+    start_time_seconds = int(start_time.timestamp())
+    end_time_seconds = int(end_time.timestamp())
+    # Cpu metrics data can take up to 240 seconds to appear
+    if (now_seconds - end_time_seconds) < DATAFLOW_METRICS_DELAY_SECONDS:
+      logging.info('Waiting for Dataflow metrics to be available (up to 3 minutes)...')
+      time.sleep(DATAFLOW_METRICS_DELAY_SECONDS - (now_seconds - end_time_seconds))
+
+    interval = TimeInterval(
+        {
+            "start_time": {"seconds": start_time_seconds},
+            "end_time": {"seconds": end_time_seconds},
+        }
+    )
+
+    api_filter = (
+      'metric.type = "dataflow.googleapis.com/job/elements_produced_count" '
+      f'AND resource.labels.project_id = "{self.project}" '
+      f'AND metric.labels.job_id = "{self.job_id}" '
+      f'AND metric.labels.ptransform = "{ptransform}" '
+    )
+
+    aggregation = Aggregation(
+        {
+            "alignment_period": {"seconds": 60},  # 1 minute
+            "per_series_aligner": Aggregation.Aligner.ALIGN_RATE,
+            # "group_by_fields": ["metric.job_id", "metric.ptransform"],
+        }
+    )
+
+    results = client.list_time_series(
+      request={
+        "name": project_name,
+        "filter": api_filter,
+        "interval": interval,
+        "view": monitoring_v3.ListTimeSeriesRequest.TimeSeriesView.FULL,
+        "aggregation": aggregation
+      }
+    )
+
+    if not results:
+      logging.warn('No monitoring data found. Unable to calculate max throughput.')
+      return None
+
+    return self._GetMaxValueFromTimeSeries(results)
+
+  def GetSubscriptionBacklogSize(self, subscription_name, interval_length=4):
```

Comment:
Does this actually get used in this class? If not I suggest moving it to the subclass that will be in the subsequent PR.